erpnext/stock/dashboard/item_dashboard.js

```diff
@@ -75,13 +75,21 @@ erpnext.stock.ItemDashboard = Class.extend({
 			this.content.find('.more').addClass('hidden');
 		}
 
-        // If not any stock in any warehouses provide a message to end user
+		// If not any stock in any warehouses provide a message to end user
 		if (context.data.length > 0) {
 			$(frappe.render_template('item_dashboard_list', context)).appendTo(this.result);
 		} else {
 			var message = __(" Currently no stock available in any warehouse")
 			$("<span class='text-muted small'>"+message+"</span>").appendTo(this.result);
 		}
+
+		if (frappe.boot.user.can_write.indexOf("Stock Entry")>=0){
+			$(".btn-move").show();
+			$(".btn-add").show();
+		}else{
+			$(".btn-move").hide();
+			$(".btn-add").hide();
+		}
```

Comment:
@vinayak-- 
Don't check in item_dashbord.js, add has_access true/false based on permission in context and then check has_access in [item_dashboard_list.html](https://github.com/frappe/erpnext/blob/develop/erpnext/stock/dashboard/item_dashboard_list.html#L42-L54)